index.js

```diff
@@ -25,27 +25,53 @@ function compose (middleware) {
   }
 
   /**
+   * Make an iterator for middleware.
+   */
+
+  middleware[Symbol.iterator] = function () {
+    const self = this
+    const length = this.length
+    let i = 0
+    let context
+    let nextFunc
+
+    return {
+      next (c, n) {
+        if (!context) context = c
+        if (!nextFunc) nextFunc = n
+
+        let fn = self[i++] || nextFunc
+        let done = i > length
+        let value
+
+        if (done) {
+          value = fn ? fn(context) : void 0
+        } else {
+          value = fn(context, this.next.bind(this))
+        }
+
+        return {
+          value,
+          done
+        }
+      }
+    }
+  }
+
+  // Alternative iteration.
+  const iter = middleware[Symbol.iterator]()
```

Comment:
i'm wondering if instead of mutating the `middleware` object, we could create a totally separate iterator. might remove the need for a bind. maybe even use a generator? not sure if that would be faster or slower.